src/hotspot/share/code/nmethod.cpp

```diff
@@ -1445,14 +1441,16 @@ nmethod::nmethod(
 #endif
     assert((data_offset() + data_end_offset) <= nmethod_size, "wrong nmethod's size: %d < %d", nmethod_size, (data_offset() + data_end_offset));
 
-    // after _scopes_pcs_offset is set
-    _pc_desc_container.reset_to(scopes_pcs_begin());
-
+    // Copy code and relocation info
     code_buffer->copy_code_and_locs_to(this);
-    // Copy contents of ScopeDescRecorder to nmethod
+    // Copy oops and metadata
     code_buffer->copy_values_to(this);
-    debug_info->copy_to(this);
     dependencies->copy_to(this);
+    // Copy PcDesc and ScopeDesc data
+    debug_info->copy_to(this);
+
+    // Create cache after PcDesc data is copied - it will be used to initialize cache
+    _pc_desc_container = new PcDescContainer(scopes_pcs_begin());
```

Comment:
It was a bug from early HotSpot days. The assert in `PcDescCache::init_to()` checks that initializing data is sentinel but before PcDesc data is copied values in allocated space could be anything. I don't understand how we don't hit this assert before in mainline. We always hit it on Aarch64 in `leyden/premain` repo.